OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -1190,10 +1192,18 @@ public virtual bool CanTarget(Actor self, Target target, List<Actor> othersAtTar
 				if (target.Type != TargetType.Terrain || (aircraft.requireForceMove && !modifiers.HasModifier(TargetModifiers.ForceMove)))
 					return false;
 
+				var location = self.World.Map.CellContaining(target.CenterPosition);
+
+				// Aircraft can be force-landed by issuing a force-move order on a clear terrain cell
+				// Cells that contain a blocking building are treated as regular force move orders, overriding
+				// selection for left-mouse orders
 				if (modifiers.HasModifier(TargetModifiers.ForceMove) && aircraft.Info.CanForceLand)
-					OrderID = "Land";
+				{
+					var building = bi.GetBuildingAt(location);
+					if (building == null || building.TraitOrDefault<Selectable>() == null || aircraft.CanLand(location, blockedByMobile: false))
```

Comment:
Hinging this on `Selectable` may be too much of a hack. We could expose this as a flag on `BuildingInfo` instead?

Reply:
What about a Foundation flag? Say, l and L for a landable x (since rest are obviously passable already).

Reply:
This is an actor-level consideration, so I don't think it makes any sense to tie it to specific cells in the footprint.